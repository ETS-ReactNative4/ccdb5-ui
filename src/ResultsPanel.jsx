import React from 'react';
import { connect } from 'react-redux'
import ActionBar from './ActionBar';
import ComplaintCard from './ComplaintCard';
import Pagination from './Pagination';
import './ResultsPanel.less';

export class ResultsPanel extends React.Component {
  render() {
    let composeClasses = 'results-panel';
    if (this.props.className) {
      composeClasses += ' ' + this.props.className;
    }

    let to = this.props.from + this.props.size;

    return (
        <section className={composeClasses}>
          <ActionBar />
          <ul className="cards-panel">
            {this.props.items
              .filter((e, i) => i >= this.props.from && i < to)
              .map(item => <ComplaintCard key={item.complaint_id}
                                          row={item} />)}
          </ul>
          <Pagination {...this.props} />
        </section>
    );
  }
}

const mapStateToProps = state => {
  return {
    from: state.query.from,
    size: state.query.size,
    items: state.results.items,
    total: state.results.total
  }
}

export default connect(mapStateToProps)(ResultsPanel)
