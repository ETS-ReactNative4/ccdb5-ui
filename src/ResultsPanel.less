@import (less) "base.less";

.results-panel {
    
    .cards-panel {
        padding: 0;
        margin: 0;
    }
}