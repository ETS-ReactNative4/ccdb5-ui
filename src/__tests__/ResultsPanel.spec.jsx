import React from 'react';
import { ResultsPanel } from '../ResultsPanel';
import renderer from 'react-test-renderer';

describe('initial state', () => {
  it('renders without crashing', () => {
    const items = [
      {
        company: 'foo',
        company_public_response: 'Closed',
        company_response: 'Closed',
        complaint_id: '1',
        complaint_what_happened: 'Lorem Ipsum',
        consumer_consent_provided: 'Yes',
        consumer_disputed: 'No',
        date_received: '2013-02-03',
        date_sent_to_company: '2013-02-03',
        issue: 'Foo',
        product: 'Bar',
        state: 'DC',
        sub_issue: 'Baz',
        sub_product: 'Qaz',
        submitted_via: 'email',
        timely: 'yes',
        zip_code: '200XX' 
      }
    ];

    const target = renderer.create(
      <ResultsPanel items={ items } from="0" size="10" />
    );

    let tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
