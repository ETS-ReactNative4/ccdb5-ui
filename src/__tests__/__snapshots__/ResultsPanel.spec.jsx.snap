exports[`component:ReactPanel displays a message when an error has occurred 1`] = `
<section
  className="results-panel">
  <summary
    className="action-bar">
    <div>
      <h2>
        Showing 
        <span>
          0
        </span>
         matches out of 
        <span>
          100
        </span>
         total complaints
      </h2>
    </div>
    <div
      className="layout-row">
      <div
        className="cf-select">
        <select
          id="choose-size"
          onChange={[Function]}
          value={10}>
          <option
            value={10}>
            Show 
            10
             results
          </option>
          <option
            value={25}>
            Show 
            25
             results
          </option>
          <option
            value={50}>
            Show 
            50
             results
          </option>
          <option
            value={100}>
            Show 
            100
             results
          </option>
        </select>
      </div>
      <div
        className="cf-select">
        <select
          id="choose-sort"
          onChange={[Function]}
          value={undefined}>
          <option
            value="created_date_desc">
            Sort by newest to oldest
          </option>
          <option
            value="created_date_asc">
            Sort by oldest to newest
          </option>
          <option
            value="relevance_desc">
            Sort by relevance
          </option>
          <option
            value="relevance_asc">
            Sort by relevance (asc)
          </option>
        </select>
      </div>
      <h5
        className="flex-all">
        <button
          className="a-btn a-btn__link"
          onClick={[Function]}>
          Export results
        </button>
      </h5>
    </div>
  </summary>
  <h2>
    There was a problem executing your search
  </h2>
</section>
`;

exports[`component:ReactPanel displays a message when the data has issues 1`] = `
<section
  className="results-panel">
  <summary
    className="action-bar">
    <div>
      <h2>
        Showing 
        <span>
          1
        </span>
         matches out of 
        <span>
          100
        </span>
         total complaints
      </h2>
    </div>
    <div
      className="layout-row">
      <div
        className="cf-select">
        <select
          id="choose-size"
          onChange={[Function]}
          value={10}>
          <option
            value={10}>
            Show 
            10
             results
          </option>
          <option
            value={25}>
            Show 
            25
             results
          </option>
          <option
            value={50}>
            Show 
            50
             results
          </option>
          <option
            value={100}>
            Show 
            100
             results
          </option>
        </select>
      </div>
      <div
        className="cf-select">
        <select
          id="choose-sort"
          onChange={[Function]}
          value={undefined}>
          <option
            value="created_date_desc">
            Sort by newest to oldest
          </option>
          <option
            value="created_date_asc">
            Sort by oldest to newest
          </option>
          <option
            value="relevance_desc">
            Sort by relevance
          </option>
          <option
            value="relevance_asc">
            Sort by relevance (asc)
          </option>
        </select>
      </div>
      <h5
        className="flex-all">
        <button
          className="a-btn a-btn__link"
          onClick={[Function]}>
          Export results
        </button>
      </h5>
    </div>
  </summary>
  <div
    className="m-notification m-notification__visible m-notification__warning">
    <span
      className="m-notification_icon cf-icon" />
    <div
      className="m-notification_content">
      <div
        className="h4 m-notification_message">
        We’re currently experiencing technical issues that have delayed the refresh of data on the Consumer Complaint Database.  We expect to refresh the data in the next few days.
      </div>
    </div>
  </div>
  <ul
    className="cards-panel">
    <li
      className="card-container">
      <div
        className="card">
        <div
          className="card-left layout-column">
          <h3
            className="to-detail">
            <a
              href="detail/1">
              1
            </a>
          </h3>
          <h5>
            Matched company name
          </h5>
          <span
            className="body-copy"
            dangerouslySetInnerHTML={
              Object {
                "__html": "foo",
              }
            } />
          <br />
          <h5>
            Company response to consumer
          </h5>
          <span
            className="body-copy"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Closed",
              }
            } />
          <br />
          <h5>
            Timely response?
          </h5>
          <span
            className="body-copy"
            dangerouslySetInnerHTML={
              Object {
                "__html": "yes",
              }
            } />
        </div>
        <div
          className="card-right layout-column">
          <div
            className="layout-row">
            <div
              className="layout-row">
              <h5>
                Date received:
              </h5>
              <span
                className="body-copy">
                <span>
                  2/3/2013
                </span>
              </span>
            </div>
            <div
              className="spacer" />
            <div
              className="layout-row">
              <h5>
                Consumer\'s state:
              </h5>
              <span
                className="body-copy"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "DC",
                  }
                } />
            </div>
          </div>
          <br />
          <h5>
            Product
          </h5>
          <h3
            dangerouslySetInnerHTML={
              Object {
                "__html": "Bar",
              }
            } />
          <div
            className="layout-row">
            <span
              className="body-copy subitem">
              Sub-product:
            </span>
            <span
              className="body-copy"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Qaz",
                }
              } />
          </div>
          <br />
          <h5>
            Issue
          </h5>
          <h3
            dangerouslySetInnerHTML={
              Object {
                "__html": "Foo",
              }
            } />
          <div
            className="layout-row">
            <span
              className="body-copy subitem">
              Sub-issue:
            </span>
            <span
              className="body-copy"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Baz",
                }
              } />
          </div>
          <br />
          <div>
            <h5>
              Consumer Complaint Narrative
            </h5>
            <span
              className="body-copy"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Lorem Ipsum",
                }
              } />
          </div>
        </div>
      </div>
    </li>
  </ul>
  <nav
    aria-label="Pagination"
    className="m-pagination"
    role="navigation">
    <button
      className="a-btn m-pagination_btn-prev"
      disabled={true}
      onClick={[Function]}>
      <span
        className="a-btn_icon a-btn_icon__on-left cf-icon cf-icon__before cf-icon-left" />
      Previous
    </button>
    <button
      className="a-btn m-pagination_btn-next"
      disabled={true}
      onClick={[Function]}>
      Next
      <span
        className="a-btn_icon a-btn_icon__on-right cf-icon cf-icon__after cf-icon-right" />
    </button>
    <form
      action=""
      className="m-pagination_form"
      onSubmit={[Function]}>
      <label
        className="m-pagination_label"
        htmlFor="m-pagination_current-page">
        Page
        <span
          className="u-visually-hidden">
          number 
          1
           out
        </span>
        <input
          className="m-pagination_current-page"
          id="m-pagination_current-page"
          inputMode="numeric"
          max={1}
          min="1"
          name="page"
          onChange={[Function]}
          pattern="[0-9]*"
          type="number"
          value={1} />
        of 
        1
      </label>
      <button
        className="a-btn a-btn__link m-pagination_btn-submit"
        id="m-pagination_btn-submit"
        type="submit">
        Go
      </button>
    </form>
  </nav>
</section>
`;

exports[`component:ReactPanel displays a message when the data is stale 1`] = `
<section
  className="results-panel">
  <summary
    className="action-bar">
    <div>
      <h2>
        Showing 
        <span>
          1
        </span>
         matches out of 
        <span>
          100
        </span>
         total complaints
      </h2>
    </div>
    <div
      className="layout-row">
      <div
        className="cf-select">
        <select
          id="choose-size"
          onChange={[Function]}
          value={10}>
          <option
            value={10}>
            Show 
            10
             results
          </option>
          <option
            value={25}>
            Show 
            25
             results
          </option>
          <option
            value={50}>
            Show 
            50
             results
          </option>
          <option
            value={100}>
            Show 
            100
             results
          </option>
        </select>
      </div>
      <div
        className="cf-select">
        <select
          id="choose-sort"
          onChange={[Function]}
          value={undefined}>
          <option
            value="created_date_desc">
            Sort by newest to oldest
          </option>
          <option
            value="created_date_asc">
            Sort by oldest to newest
          </option>
          <option
            value="relevance_desc">
            Sort by relevance
          </option>
          <option
            value="relevance_asc">
            Sort by relevance (asc)
          </option>
        </select>
      </div>
      <h5
        className="flex-all">
        <button
          className="a-btn a-btn__link"
          onClick={[Function]}>
          Export results
        </button>
      </h5>
    </div>
  </summary>
  <div
    className="m-notification m-notification__visible m-notification__warning">
    <span
      className="m-notification_icon cf-icon" />
    <div
      className="m-notification_content">
      <div
        className="h4 m-notification_message">
        We’re currently experiencing technical issues that have delayed the refresh of consumer complaint narratives on the Consumer Complaint Database.  We expect to refresh the data in the next few days.
      </div>
    </div>
  </div>
  <ul
    className="cards-panel">
    <li
      className="card-container">
      <div
        className="card">
        <div
          className="card-left layout-column">
          <h3
            className="to-detail">
            <a
              href="detail/1">
              1
            </a>
          </h3>
          <h5>
            Matched company name
          </h5>
          <span
            className="body-copy"
            dangerouslySetInnerHTML={
              Object {
                "__html": "foo",
              }
            } />
          <br />
          <h5>
            Company response to consumer
          </h5>
          <span
            className="body-copy"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Closed",
              }
            } />
          <br />
          <h5>
            Timely response?
          </h5>
          <span
            className="body-copy"
            dangerouslySetInnerHTML={
              Object {
                "__html": "yes",
              }
            } />
        </div>
        <div
          className="card-right layout-column">
          <div
            className="layout-row">
            <div
              className="layout-row">
              <h5>
                Date received:
              </h5>
              <span
                className="body-copy">
                <span>
                  2/3/2013
                </span>
              </span>
            </div>
            <div
              className="spacer" />
            <div
              className="layout-row">
              <h5>
                Consumer\'s state:
              </h5>
              <span
                className="body-copy"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "DC",
                  }
                } />
            </div>
          </div>
          <br />
          <h5>
            Product
          </h5>
          <h3
            dangerouslySetInnerHTML={
              Object {
                "__html": "Bar",
              }
            } />
          <div
            className="layout-row">
            <span
              className="body-copy subitem">
              Sub-product:
            </span>
            <span
              className="body-copy"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Qaz",
                }
              } />
          </div>
          <br />
          <h5>
            Issue
          </h5>
          <h3
            dangerouslySetInnerHTML={
              Object {
                "__html": "Foo",
              }
            } />
          <div
            className="layout-row">
            <span
              className="body-copy subitem">
              Sub-issue:
            </span>
            <span
              className="body-copy"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Baz",
                }
              } />
          </div>
          <br />
          <div>
            <h5>
              Consumer Complaint Narrative
            </h5>
            <span
              className="body-copy"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Lorem Ipsum",
                }
              } />
          </div>
        </div>
      </div>
    </li>
  </ul>
  <nav
    aria-label="Pagination"
    className="m-pagination"
    role="navigation">
    <button
      className="a-btn m-pagination_btn-prev"
      disabled={true}
      onClick={[Function]}>
      <span
        className="a-btn_icon a-btn_icon__on-left cf-icon cf-icon__before cf-icon-left" />
      Previous
    </button>
    <button
      className="a-btn m-pagination_btn-next"
      disabled={true}
      onClick={[Function]}>
      Next
      <span
        className="a-btn_icon a-btn_icon__on-right cf-icon cf-icon__after cf-icon-right" />
    </button>
    <form
      action=""
      className="m-pagination_form"
      onSubmit={[Function]}>
      <label
        className="m-pagination_label"
        htmlFor="m-pagination_current-page">
        Page
        <span
          className="u-visually-hidden">
          number 
          1
           out
        </span>
        <input
          className="m-pagination_current-page"
          id="m-pagination_current-page"
          inputMode="numeric"
          max={1}
          min="1"
          name="page"
          onChange={[Function]}
          pattern="[0-9]*"
          type="number"
          value={1} />
        of 
        1
      </label>
      <button
        className="a-btn a-btn__link m-pagination_btn-submit"
        id="m-pagination_btn-submit"
        type="submit">
        Go
      </button>
    </form>
  </nav>
</section>
`;

exports[`component:ReactPanel displays a message when there are no results 1`] = `
<section
  className="results-panel">
  <summary
    className="action-bar">
    <div>
      <h2>
        Showing 
        <span>
          0
        </span>
         matches out of 
        <span>
          100
        </span>
         total complaints
      </h2>
    </div>
    <div
      className="layout-row">
      <div
        className="cf-select">
        <select
          id="choose-size"
          onChange={[Function]}
          value={10}>
          <option
            value={10}>
            Show 
            10
             results
          </option>
          <option
            value={25}>
            Show 
            25
             results
          </option>
          <option
            value={50}>
            Show 
            50
             results
          </option>
          <option
            value={100}>
            Show 
            100
             results
          </option>
        </select>
      </div>
      <div
        className="cf-select">
        <select
          id="choose-sort"
          onChange={[Function]}
          value={undefined}>
          <option
            value="created_date_desc">
            Sort by newest to oldest
          </option>
          <option
            value="created_date_asc">
            Sort by oldest to newest
          </option>
          <option
            value="relevance_desc">
            Sort by relevance
          </option>
          <option
            value="relevance_asc">
            Sort by relevance (asc)
          </option>
        </select>
      </div>
      <h5
        className="flex-all">
        <button
          className="a-btn a-btn__link"
          onClick={[Function]}>
          Export results
        </button>
      </h5>
    </div>
  </summary>
  <h2>
    No results were found for your search
  </h2>
</section>
`;

exports[`component:ReactPanel renders without crashing 1`] = `
<section
  className="results-panel">
  <summary
    className="action-bar">
    <div>
      <h2>
        Showing 
        <span>
          1
        </span>
         matches out of 
        <span>
          100
        </span>
         total complaints
      </h2>
    </div>
    <div
      className="layout-row">
      <div
        className="cf-select">
        <select
          id="choose-size"
          onChange={[Function]}
          value={10}>
          <option
            value={10}>
            Show 
            10
             results
          </option>
          <option
            value={25}>
            Show 
            25
             results
          </option>
          <option
            value={50}>
            Show 
            50
             results
          </option>
          <option
            value={100}>
            Show 
            100
             results
          </option>
        </select>
      </div>
      <div
        className="cf-select">
        <select
          id="choose-sort"
          onChange={[Function]}
          value={undefined}>
          <option
            value="created_date_desc">
            Sort by newest to oldest
          </option>
          <option
            value="created_date_asc">
            Sort by oldest to newest
          </option>
          <option
            value="relevance_desc">
            Sort by relevance
          </option>
          <option
            value="relevance_asc">
            Sort by relevance (asc)
          </option>
        </select>
      </div>
      <h5
        className="flex-all">
        <button
          className="a-btn a-btn__link"
          onClick={[Function]}>
          Export results
        </button>
      </h5>
    </div>
  </summary>
  <ul
    className="cards-panel">
    <li
      className="card-container">
      <div
        className="card">
        <div
          className="card-left layout-column">
          <h3
            className="to-detail">
            <a
              href="detail/1">
              1
            </a>
          </h3>
          <h5>
            Matched company name
          </h5>
          <span
            className="body-copy"
            dangerouslySetInnerHTML={
              Object {
                "__html": "foo",
              }
            } />
          <br />
          <h5>
            Company response to consumer
          </h5>
          <span
            className="body-copy"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Closed",
              }
            } />
          <br />
          <h5>
            Timely response?
          </h5>
          <span
            className="body-copy"
            dangerouslySetInnerHTML={
              Object {
                "__html": "yes",
              }
            } />
        </div>
        <div
          className="card-right layout-column">
          <div
            className="layout-row">
            <div
              className="layout-row">
              <h5>
                Date received:
              </h5>
              <span
                className="body-copy">
                <span>
                  2/3/2013
                </span>
              </span>
            </div>
            <div
              className="spacer" />
            <div
              className="layout-row">
              <h5>
                Consumer\'s state:
              </h5>
              <span
                className="body-copy"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "DC",
                  }
                } />
            </div>
          </div>
          <br />
          <h5>
            Product
          </h5>
          <h3
            dangerouslySetInnerHTML={
              Object {
                "__html": "Bar",
              }
            } />
          <div
            className="layout-row">
            <span
              className="body-copy subitem">
              Sub-product:
            </span>
            <span
              className="body-copy"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Qaz",
                }
              } />
          </div>
          <br />
          <h5>
            Issue
          </h5>
          <h3
            dangerouslySetInnerHTML={
              Object {
                "__html": "Foo",
              }
            } />
          <div
            className="layout-row">
            <span
              className="body-copy subitem">
              Sub-issue:
            </span>
            <span
              className="body-copy"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Baz",
                }
              } />
          </div>
          <br />
          <div>
            <h5>
              Consumer Complaint Narrative
            </h5>
            <span
              className="body-copy"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Lorem Ipsum",
                }
              } />
          </div>
        </div>
      </div>
    </li>
  </ul>
  <nav
    aria-label="Pagination"
    className="m-pagination"
    role="navigation">
    <button
      className="a-btn m-pagination_btn-prev"
      disabled={true}
      onClick={[Function]}>
      <span
        className="a-btn_icon a-btn_icon__on-left cf-icon cf-icon__before cf-icon-left" />
      Previous
    </button>
    <button
      className="a-btn m-pagination_btn-next"
      disabled={true}
      onClick={[Function]}>
      Next
      <span
        className="a-btn_icon a-btn_icon__on-right cf-icon cf-icon__after cf-icon-right" />
    </button>
    <form
      action=""
      className="m-pagination_form"
      onSubmit={[Function]}>
      <label
        className="m-pagination_label"
        htmlFor="m-pagination_current-page">
        Page
        <span
          className="u-visually-hidden">
          number 
          1
           out
        </span>
        <input
          className="m-pagination_current-page"
          id="m-pagination_current-page"
          inputMode="numeric"
          max={1}
          min="1"
          name="page"
          onChange={[Function]}
          pattern="[0-9]*"
          type="number"
          value={1} />
        of 
        1
      </label>
      <button
        className="a-btn a-btn__link m-pagination_btn-submit"
        id="m-pagination_btn-submit"
        type="submit">
        Go
      </button>
    </form>
  </nav>
</section>
`;
